### PR TITLE
database write prohibition integration: mongodb and mysql action

### LINKED ISSUE #1461: Add the database write prohibition integration test
### What would you like to be enhanced?

1、Add the database write prohibition integration test.

### Why is this needed?

1、Prevent function exceptions caused by subsequent plugin code changes.

### PR DESCRIPTION
**Pull request by daizhenyu**
**What type of PR is this?**

Integration Test

**What this PR does / why we need it?**

database write prohibition plugin integration test : mongodb and mysql action

**Which issue(s) this PR fixes？**

Fixes #1461 

**Does this PR introduce a user-facing change?**

No

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/huaweicloud/Sermant/issues) related with this PR before you start working on it.
- [x] Make sure you have squashed your change to one single commit.
- [x] GitHub Actions works fine in this PR.

### CODE REVIEW

File: .github/actions/scenarios/database-write-prohibition/mongodb/action.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        nohup mongo test &
+    - name: package demos
+      shell: bash
+      run: |
+        mvn package -Dmongodb.version=${{ matrix.mongodbClientVersion }} -DskipTests -Pmongodb-test --file \
+        sermant-integration-tests/database-write-prohibition-test/pom.xml
+    - name: start mongodb demo
+      shell: bash
+      run: |
+        nohup java -javaagent:sermant-agent-${{ env.sermantVersion }}/agent/sermant-agent.jar=appName=default -jar \
```
Comment (lilai23) [inline .github/actions/scenarios/database-write-prohibition/mongodb/action.yml]: remove appName=default

File: .github/actions/scenarios/database-write-prohibition/mysql/action.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y mysql-server
+        sudo systemctl start mysql.service
+        config_file="/etc/mysql/debian.cnf"
+        user=$(sudo awk -F '=' '/user/ {print $2; exit}' "/etc/mysql/debian.cnf" | tr -d '[:space:]')
+        password=$(sudo awk -F '=' '/password/ {print $2; exit}' "/etc/mysql/debian.cnf" | tr -d '[:space:]')
+        sudo mysql -u $user -p$password -e "CREATE DATABASE IF NOT EXISTS test;"
+        echo "MYSQL_USER=$user" >> $GITHUB_ENV
+        echo "MYSQL_PASSWORD=$password" >> $GITHUB_ENV
```
Comment (lilai23) [inline .github/actions/scenarios/database-write-prohibition/mysql/action.yml:20]: need to create env in this repository？

File: .github/actions/scenarios/database-write-prohibition/mysql/action.yml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    - name: package demos
+      shell: bash
+      run: |
+        mvn package -Dmariadb.version=${{ matrix.mysqlClientVersion }} -DskipTests -Pmysql-test --file \
+        sermant-integration-tests/database-write-prohibition-test/pom.xml
+    - name: start mysql demo
+      shell: bash
+      run: |
+        nohup java -javaagent:sermant-agent-${{ env.sermantVersion }}/agent/sermant-agent.jar=appName=default -jar \
```
Comment (lilai23) [inline .github/actions/scenarios/database-write-prohibition/mysql/action.yml]: remove appName=default

File: .github/workflows/database_write_prohibition_integration_test.yml [hunk 4]
```diff
@@ -54,4 +54,61 @@ jobs:
...
+          - mongodbClientVersion: "3.9.1"
+          - mongodbClientVersion: "3.10.2"
+          - mongodbClientVersion: "3.11.3"
+          - mongodbClientVersion: "3.12.14"
+          - mongodbClientVersion: "4.0.6"
+          - mongodbClientVersion: "4.2.3"
+          - mongodbClientVersion: "4.5.1"
+          - mongodbClientVersion: "4.7.2"
+          - mongodbClientVersion: "4.9.1"
+          - mongodbClientVersion: "4.11.1"
```
Comment (lilai23) [inline .github/workflows/database_write_prohibition_integration_test.yml]: cover every x.x version in integration test

File: .github/actions/scenarios/database-write-prohibition/mysql/action.yml (hunk 2 above)
Comment (daizhenyu) [reply .github/actions/scenarios/database-write-prohibition/mysql/action.yml:20]: To specify the user and password when starting the mysql demo in the next step

File: .github/actions/scenarios/database-write-prohibition/mongodb/action.yml (hunk 1 above)
Comment (daizhenyu) [reply .github/actions/scenarios/database-write-prohibition/mongodb/action.yml]: resolved

File: .github/actions/scenarios/database-write-prohibition/mysql/action.yml (hunk 3 above)
Comment (daizhenyu) [reply .github/actions/scenarios/database-write-prohibition/mysql/action.yml]: resolved

File: .github/workflows/database_write_prohibition_integration_test.yml (hunk 4 above)
Comment (daizhenyu) [reply .github/workflows/database_write_prohibition_integration_test.yml]: resolved

**Review by lilai23 — DISMISSED** [review]

File: .github/actions/scenarios/database-write-prohibition/mysql/action.yml [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+      uses: ./.github/actions/common/entry
+      with:
+        log-dir: ./logs/databse-write-prohibition/mysql
+    - name: start mysql
+      shell: bash
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y mysql-server
+        sudo systemctl start mysql.service
+        config_file="/etc/mysql/debian.cnf"
```
Comment (hanbingleixue) [inline .github/actions/scenarios/database-write-prohibition/mysql/action.yml]: The config_file variable is not used.

File: .github/actions/scenarios/database-write-prohibition/mysql/action.yml [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    - name: exit
+      if: always()
+      uses: ./.github/actions/common/exit
+      with:
+        processor-keyword: mariadb
+    - name: if failure then upload error log
+      uses: actions/upload-artifact@v3
+      if: ${{ failure() || cancelled() }}
+      with:
+        name: (${{ github.job }})-database-write-prohibition-mariadb-(${{ matrix.mysqlClientVersion }}-logs
```
Comment (hanbingleixue) [inline .github/actions/scenarios/database-write-prohibition/mysql/action.yml]: Parentheses in (${{matrix.mysqlClientVersion}}-logs are not closed

File: .github/actions/scenarios/database-write-prohibition/mysql/action.yml (hunk 5 above)
Comment (daizhenyu) [reply .github/actions/scenarios/database-write-prohibition/mysql/action.yml]: resolved

File: .github/actions/scenarios/database-write-prohibition/mysql/action.yml (hunk 6 above)
Comment (daizhenyu) [reply .github/actions/scenarios/database-write-prohibition/mysql/action.yml]: resolved

**Review by hanbingleixue — APPROVED** [review]

**Review by lilai23 — APPROVED** [review]

**Pull request merged**